tests/integration/sharding_tests/test_sharding_backups.py

```diff
@@ -0,0 +1,127 @@
+    )
+    await ops_test.model.deploy(
+        my_charm, num_units=2, config={"role": "shard"}, application_name=SHARD_ONE_APP_NAME
+    )
+    await ops_test.model.deploy(
+        my_charm, num_units=1, config={"role": "shard"}, application_name=SHARD_TWO_APP_NAME
+    )
+
+    # deploy the s3 integrator charm
+    await ops_test.model.deploy(S3_APP_NAME, channel="edge")
+
+    await ops_test.model.wait_for_idle(
+        apps=[S3_APP_NAME, CONFIG_SERVER_APP_NAME, SHARD_ONE_APP_NAME, SHARD_TWO_APP_NAME],
+        idle_period=20,
+        raise_on_blocked=False,
+        timeout=TIMEOUT,
+        raise_on_error=False,
+    )
+
+
+@pytest.mark.group(1)
+async def test_set_credentials_in_cluster(ops_test: OpsTest, github_secrets) -> None:
+    """Tests that sharded cluster can be configured for s3 configurations."""
+    await backup_helpers.set_credentials(ops_test, github_secrets, cloud="AWS")
+    choices = string.ascii_letters + string.digits
+    unique_path = "".join([secrets.choice(choices) for _ in range(4)])
+    configuration_parameters = {
+        "bucket": "data-charms-testing",
+        "path": f"mongodb-vm/test-{unique_path}",
+        "endpoint": "https://s3.amazonaws.com",
+        "region": "us-east-1",
+    }
+
+    # apply new configuration options
+    await ops_test.model.applications[S3_APP_NAME].set_config(configuration_parameters)
+    await ops_test.model.wait_for_idle(apps=[S3_APP_NAME], status="active", timeout=TIMEOUT)
+
+    # provide config-server to entire cluster and s3-integrator to config-server - integrations
+    # made in succession to test race conditions.
+    await ops_test.model.integrate(
+        f"{S3_APP_NAME}:{S3_REL_NAME}",
+        f"{CONFIG_SERVER_APP_NAME}:{S3_REL_NAME}",
+    )
+    await ops_test.model.integrate(
+        f"{SHARD_ONE_APP_NAME}:{SHARD_REL_NAME}",
+        f"{CONFIG_SERVER_APP_NAME}:{CONFIG_SERVER_REL_NAME}",
+    )
+    await ops_test.model.integrate(
+        f"{SHARD_TWO_APP_NAME}:{SHARD_REL_NAME}",
+        f"{CONFIG_SERVER_APP_NAME}:{CONFIG_SERVER_REL_NAME}",
+    )
+
+    await ops_test.model.wait_for_idle(
+        apps=[
+            CONFIG_SERVER_APP_NAME,
+            SHARD_ONE_APP_NAME,
+            SHARD_TWO_APP_NAME,
+        ],
+        idle_period=20,
+        status="active",
+        timeout=TIMEOUT,
+    )
+
+
+@pytest.mark.group(1)
+async def test_create_and_list_backups_in_cluster(ops_test: OpsTest, github_secrets) -> None:
+    """Tests that sharded cluster can successfully create and list backups."""
+    leader_unit = await backup_helpers.get_leader_unit(
+        ops_test, db_app_name=CONFIG_SERVER_APP_NAME
+    )
+    await backup_helpers.set_credentials(ops_test, github_secrets, cloud="AWS")
+    # verify backup list works
+    action = await leader_unit.run_action(action_name="list-backups")
+    list_result = await action.wait()
+    backups = list_result.results["backups"]
+    assert backups, "backups not outputted"
+
```

Comment:
I think in a separate PR, we need to test that backups actually work across shards to avoid situations when the backup is restored only to one shard. Also, we should test that the backup can be restored to the new cluster.
The test can have the following logic:
1. write some data into a shared collection.
2. check the amount of data on each shard
3. make a backup
4. write more data.
5. check the amount of data grown on each shard.
6. stop writing. make a restore
7. check the amount of data on each shard; should be equal to step 2

Reply:
Thank you! we'll track both tests on [DPE-3608](https://warthogs.atlassian.net/browse/DPE-3608) and [DPE-3609](https://warthogs.atlassian.net/browse/DPE-3609)

Reply:
We have a JIRA task for restores [here](https://warthogs.atlassian.net/browse/DPE-3501)